conftest.py

```diff
@@ -21,7 +21,7 @@ def pytest_addoption(parser):
         help="If provided, will not annihilate the data generated.")
 
 
-@pytest.fixture(scope="module")
+@pytest.fixture()
 def workspace(request):
     # TODO: Use pytest.config.option('--workspace')
     test_workspace = tempfile.mkdtemp()
```

Comment:
the `workspace` fixture has a finalizer on the request, L34. the lifetime of the request ends each time a set of parameters are served to the test. Therefore each iteration for each test gets a fresh directory (courtesy of line 27) and gets cleaned up on test-instance-exit by L32, via the `addfinalizer` callback.